GW2EIEvtcParser/EIData/Buffs/BuffSimulators/BuffSimulationItems/BuffSimulationItemBase.cs

```diff
@@ -1,132 +1,130 @@
+    public override void OverrideEnd(long end)
+    {
+        this.End = end;
+    }
 
-        public override void OverrideEnd(long end)
-        {
-            Duration = Math.Min(Math.Max(end - Start, 0), Duration);
-        }
+    public override int GetActiveStacks()
+    {
+        return GetStacks();
+    }
 
-        public override int GetActiveStacks()
-        {
-            return GetStacks();
-        }
+    public override int GetStacks()
+    {
+        return 1;
+    }
 
-        public override int GetStacks()
-        {
-            return 1;
-        }
+    public override int GetActiveStacks(AbstractSingleActor actor)
+    {
+        return GetStacks(actor);
+    }
 
-        public override int GetActiveStacks(AbstractSingleActor actor)
-        {
-            return GetStacks(actor);
-        }
+    public override int GetStacks(AbstractSingleActor actor)
+    {
+        return GetActiveSources().Any(x => x == actor.AgentItem) ? 1 : 0;
+    }
 
-        public override int GetStacks(AbstractSingleActor actor)
-        {
-            if (GetActiveSources().Any(x => x == actor.AgentItem))
-            {
-                return 1;
-            }
-            return 0;
-        }
+    public override IEnumerable<long> GetActualDurationPerStack()
+    {
+        return [ GetActualDuration() ];
+    }
 
-        public override IReadOnlyList<long> GetActualDurationPerStack()
-        {
-            return new List<long>() { GetActualDuration() };
-        }
+    public override long GetActualDuration()
+    {
+        return _totalDuration;
+    }
 
-        public override long GetActualDuration()
+    public override IEnumerable<AgentItem> GetSources()
+    {
+        return [ _buffStackItem.Src ];
+    }
+
+    public override IEnumerable<AgentItem> GetActiveSources()
+    {
+        return GetSources();
+    }
+
+    public override void SetBuffDistributionItem(BuffDistribution distribs, long start, long end, long buffID)
+    {
+        long cDur = GetClampedDuration(start, end);
+        if (cDur == 0)
         {
-            return _totalDuration;
+            return;
         }
 
-        public override IReadOnlyList<AgentItem> GetSources()
+        Dictionary<AgentItem, BuffDistributionItem> distrib = distribs.GetDistrib(buffID);
+        AgentItem agent = _src;
+        AgentItem seedAgent = _buffStackItem.SeedSrc;
+        if (distrib.TryGetValue(agent, out BuffDistributionItem toModify))
         {
-            return new List<AgentItem>() { _src };
+            toModify.IncrementValue(cDur);
         }
-
-        public override IReadOnlyList<AgentItem> GetActiveSources()
+        else
         {
-            return GetSources();
+            distrib.Add(agent, new BuffDistributionItem(
+                cDur,
+                0, 0, 0, 0, 0));
         }
 
-        public override void SetBuffDistributionItem(BuffDistribution distribs, long start, long end, long buffID)
+        if (_buffStackItem.IsExtension)
```

Comment:
This won't work properly, StackID and SeedSrc are the only attributes that are constant during a BuffStackItem's lifecycle, for the same reason Start, Duration, Src and TotalDuration have to be stored, so does IsExtension.